packages/@react-spectrum/s2/src/utils.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {useMediaQuery} from '@react-spectrum/utils';
+
+export function useIsMobileDevice(): boolean {
+  return useMediaQuery('(max-width: 640px)');
```

Comment:
Since this is being used to determine the scale, shouldn't it look for `@media not ((hover: hover) and (pointer: fine))`, which is what the style macro uses for that?

Reply:
I considered doing that but I figured we'd want to follow a similar logic style that we had in S1 in preparation for Trays aka we would check screen size. If we use the touch media query that the style macro uses, that would match large screen touch devices right?

Reply:
But this is being used to determine pixel sizes not whether to show a tray right? The style macro uses the media query I posted to do the same thing, ie scale up dimensions by 25%. Whether to show a tray is indeed based on screen size but that's different from touch scale, which might also occur on large screen devices like iPads or laptops.

Reply:
ahh right, makes sense, guess I conflated the two. I'll make a separate `useScale` that uses the media query and use it for table